### PR TITLE
Issue 18

### PR DESCRIPTION
**Pull request by Damdoshi**

### CODE REVIEW

File: sources/examples/8x32_led_matrix_max7219/8x32_led_matrix_max7219.ino [hunk 1]
```diff
@@ -8,84 +8,92 @@
...
-  ball->x = random(matrix.getScreenWidth());
-  ball->y = random(matrix.getScreenHeight());
-  ball->xSpeed = random(-10, 10) / 10.0;
-  ball->ySpeed = random(-10, 10) / 10.0;
-}
-
+// This function is only called at startup once
 void setup()
 {
+  // Allow the matrix to work correctly
```
Comment (monpierrelisa) [inline sources/examples/8x32_led_matrix_max7219/8x32_led_matrix_max7219.ino:26]: // Initialization of the matrix, allow it to work correctly

File: sources/examples/8x32_led_matrix_max7219/8x32_led_matrix_max7219.ino [hunk 2]
```diff
@@ -8,84 +8,92 @@
...
-
-void setupBall(t_ball * ball)
-{
-  ball->x = random(matrix.getScreenWidth());
-  ball->y = random(matrix.getScreenHeight());
-  ball->xSpeed = random(-10, 10) / 10.0;
-  ball->ySpeed = random(-10, 10) / 10.0;
-}
-
+// This function is only called at startup once
```
Comment (monpierrelisa) [inline sources/examples/8x32_led_matrix_max7219/8x32_led_matrix_max7219.ino:23]: This function is only called once at startup to set up the demo

File: sources/examples/8x32_led_matrix_max7219/8x32_led_matrix_max7219.ino [hunk 3]
```diff
@@ -8,84 +8,92 @@
...
-  
+
+  // Set a seed from an analog entry so the random generator generates differents values at each startup
   randomSeed(analogRead(0));
-  for (uint8_t i = 0; i < NbBalls; ++i)
-  {
-    setupBall(&balls[i]);
-  }
 
+  // Set the horizontal ball position somewhere between 0 and the width of the screen randombly
```
Comment (monpierrelisa) [inline sources/examples/8x32_led_matrix_max7219/8x32_led_matrix_max7219.ino:34]: ramdomly

File: sources/examples/8x32_led_matrix_max7219/8x32_led_matrix_max7219.ino [hunk 4]
```diff
@@ -8,84 +8,92 @@
...
+  ballX = random(matrix.getScreenWidth());
+  // Set the vertical ball position somewhere between 0 and the height of the screen randombly
```
Comment (monpierrelisa) [inline sources/examples/8x32_led_matrix_max7219/8x32_led_matrix_max7219.ino:36]: ramdomly

File: sources/examples/8x32_led_matrix_max7219/8x32_led_matrix_max7219.ino [hunk 5]
```diff
@@ -8,84 +8,92 @@
...
+  ballY = random(matrix.getScreenHeight());
+  // Set an horizontal and vertical speed between -1 and 1, but not inside -0.2 0.2 so the
+  // ball does not follow perfectly horizontal or vertical lines
+  while (fabs(ballXSpeed = random(-10, 10) / 10.0) < 0.2);
+  while (fabs(ballYSpeed = random(-10, 10) / 10.0) < 0.2);
+
+  // Set different brightness levels of matrix, so a little variety appeirs when balls move from
```
Comment (monpierrelisa) [inline sources/examples/8x32_led_matrix_max7219/8x32_led_matrix_max7219.ino:43]: appears

File: sources/examples/8x32_led_matrix_max7219/8x32_led_matrix_max7219.ino [hunk 6]
```diff
@@ -8,84 +8,92 @@
...
 
-void displayBall(t_ball * ball, uint8_t color)
+void displayBall(uint8_t color)
 {
-  matrix.setPixel(ball->x, ball->y, color);
+  // A ball is a single dot
+  matrix.setPixel(ballX, ballY, color);
 }
 
+// This function is called repeatidly
```
Comment (monpierrelisa) [inline sources/examples/8x32_led_matrix_max7219/8x32_led_matrix_max7219.ino:76]: repeatedly

**Review by monpierrelisa — CHANGES_REQUESTED** [review]

File: sources/examples/8x32_led_matrix_max7219/8x32_led_matrix_max7219.ino (hunk 3 above)
Comment (Damdoshi) [reply sources/examples/8x32_led_matrix_max7219/8x32_led_matrix_max7219.ino:34]: randomly

**Review by monpierrelisa — APPROVED** [review]

**Pull request merged**